GiniVision/Classes/Core/CameraViewController.swift

```diff
@@ -614,15 +651,56 @@ extension CameraViewController {
     }
 }
 
+// MARK: - DocumentPickerCoordinatorDelegate
+
+extension CameraViewController: DocumentPickerCoordinatorDelegate {
+
+    func documentPicker(_ coordinator: DocumentPickerCoordinator,
+                        didPick documents: [GiniVisionDocument],
+                        from: DocumentPickerType,
+                        completion: DocumentPickerCompletion?) {
+        let loadingView = addValidationLoadingView()
+        self.validate(importedDocuments: documents) { validatedDocuments in
+            loadingView.removeFromSuperview()
+            let elementsWithError = validatedDocuments.filter { $0.1 != nil}
+            if let firstElement = elementsWithError.first,
+                let error = firstElement.1,
+                (!self.giniConfiguration.multipageEnabled || firstElement.0.type != .image) {
+                if let completion = completion {
+                    completion(nil) {
+                        self.showErrorDialog(for: error)
+                    }
+                } else {
+                    self.showErrorDialog(for: error)
+                }
+            } else {
```

Comment:
This is a good place to **not** have this huge closure, but separate it logically. Here, you handle both the error and processing case. Just have two methods.
If there was an error -> `processError`
Else -> `processDocuments`